lib/DDG/Spice/Transit/Switzerland.pm

```diff
@@ -0,0 +1,58 @@
+package DDG::Spice::Transit::Switzerland;
+# ABSTRACT: Returns train times between stations in Switzerland (and some major European cities)
+
+use strict;
+use utf8;
+use DDG::Spice;
+
+primary_example_queries "next train to geneva from paris";
+secondary_example_queries "train times zurich airport to basel";
+description "Lookup the next Swiss train going your way";
+name "Swiss trains";
+source "http://transport.opendata.ch/";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Transit/Switzerland.pm";
+topics "everyday";
+category "time_sensitive";
+attribution twitter => ['ourmaninjapan', 'Daniel Davis'],
+            github  => ['tagawa', 'Daniel Davis'],
+            web     => ['http://daniemon.com/', 'Daniel Davis'];
+
+spice to => 'http://transport.opendata.ch/v1/connections?from=$1&to=$2';
+spice from => '(.*)/(.*)';
+spice wrap_jsonp_callback => 1;
+spice proxy_cache_valid => "418 1d";
+
+#load a list of stops so we don't trigger this if we don't get train stops
+#(the triggers are similar to SEPTA's)
+my @stops = share('stops.txt')->slurp;
+
+#check if the stop name is in the list of stops
+#(using the same matching algorithm as the backend)
+sub is_stop {
```

Comment:
This is pretty inefficient. We've been using YAML files in lots of Goodies with `use YAML::XS qw(load)`

If we convert stops.text to valid YAML it means we'll be able to slurp the triggers into a hash. That way we can use a hash lookup to check if the candidate location is valid. Much faster than looping over 2000+ stops!